core/store/src/trie/shard_tries.rs

```diff
@@ -284,6 +284,7 @@ impl WrappedTrieChanges {
                 "Resharding changes must never be finalized."
             );
 
+            store_update.apply_change_to_flat_state(&change_with_trie_key);
```

Comment:
Doesn’t this break when fork happens?

Reply:
Yeah, this PR doesn't work with forks and solves the simpler case. I'll update the description